crates/adapters/src/server/mod.rs

```diff
@@ -151,6 +153,36 @@ pub struct ServerArgs {
     default_port: Option<u16>,
 }
 
+pub trait CircuitBuilderFunc:
+    FnOnce(&mut RootCircuit) -> Result<Catalog, anyhow::Error> + DynClone + Send + 'static
+{
+}
+
+impl<F> CircuitBuilderFunc for F where
+    F: FnOnce(&mut RootCircuit) -> Result<Catalog, anyhow::Error> + Clone + Send + 'static
+{
+}
+
+dyn_clone::clone_trait_object! {CircuitBuilderFunc}
+
+/// A version of Runtime::init_circuit optimized for use in compiler-generated code.
+///
+/// This function does not have any generic arguments, and so will be precompiled as part
+/// of the adapters crate, pulling all scheduler code with it.  It reduces the size
+/// of LLVM code for the main generated crate from 126K to 35K lines.
+pub fn init_circuit(
```

Comment:
but this function is in the wrong namespace... who can guess to call a function from adapters to init the circuit?

Reply:
The compiler :) It's used together with server_main in the same file. "Normal" users shouldn't need it.